src/SerializerGenerator.php

```diff
@@ -78,19 +78,28 @@ public function generate(Builder $metadataBuilder): void
             foreach ($classToGenerate as $groupCombination) {
                 $className = $classToGenerate->getClassName();
                 foreach ($groupCombination->getVersions() as $version) {
+                    $groups = $groupCombination->getGroups();
                     if ('' === $version) {
-                        $metadata = $metadataBuilder->build($className, [
-                            new PreferredReducer(),
-                            new TakeBestReducer(),
-                        ]);
-                        $this->writeFile($className, null, $groupCombination->getGroups(), $metadata);
+                        if ([] === $groups) {
+                            $metadata = $metadataBuilder->build($className, [
+                                new PreferredReducer(),
+                                new TakeBestReducer(),
+                            ]);
+                            $this->writeFile($className, null, [], $metadata);
+                        } else {
+                            $metadata = $metadataBuilder->build($className, [
+                                new GroupReducer($groups),
+                                new TakeBestReducer(),
```

Comment:
i think the PreferredReducer should still occur in this list. otherwise "preferred" to select between different versions will not work anymore.